### DISCUSSION TITLE
How to get a 422 response for a missing list field

### DISCUSSION THREAD

**Discussion by PetitLepton** [fastapi/fastapi#10163] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typing
import fastapi

app = fastapi.FastAPI()

@app.get("/test")
def test(a: str, b: typing.Annotated[list[str], fastapi.Query()]) -> None:
    pass
```

### Description

Hi, I am trying to get a 422 response on a missing query parameter which is a list. While single-valued parameter automatically manage missing field, it seems that an expected list leads to a 500 error related to `pydantic` (the following is a request on `/test` without parameters).
```python
ValueError: [TypeError("'pydantic_core._pydantic_core.PydanticUndefinedType' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]
```
I can tweak the code a bit with `typing.Annotated[list[str], fastapi.Query(min_length=1)] = []` to get the 422 but it looks a bit weird as the details do not explain that the field is required
```JSON
{
  "detail": [
    {
      "type": "missing",
      "loc": [
        "query",
        "a"
      ],
      "msg": "Field required",
      "input": null,
      "url": "https://errors.pydantic.dev/2.3/v/missing"
    },
    {
      "type": "too_short",
      "loc": [
        "query",
        "b"
      ],
      "msg": "List should have at least 1 item after validation, not 0",
      "input": [],
      "ctx": {
        "field_type": "List",
        "min_length": 1,
        "actual_length": 0
      },
      "url": "https://errors.pydantic.dev/2.3/v/too_short"
    }
  ]
}
``` 

Thanks for your help.

### Operating System

Linux

### FastAPI Version

0.103.0

### Pydantic Version

2.3.0

### Python Version

Python 3.10.10

**Comment by 201RichK** [comment]
i think you need to declare a pydantic schema and pass it to your list type

**Reply by PetitLepton** [reply]
Hi, thanks for your reply. I don't think that this solves the issue. FastAPI still tries to parse the missing parameter.

**Comment by PetitLepton** [comment, marked as answer]
Actually, I just found out that there was a related issue, #9920. Thanks!
